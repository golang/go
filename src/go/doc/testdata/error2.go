// Copyright 2012 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package error2

type I0 interface {
	// When embedded, the locally-declared error interface
	// is only visible if all declarations are shown.
	error
}

type T0 struct {
	ExportedField interface {
		// error should not be visible
		error
	}
}

type S0 struct {
	// In struct types, an embedded error must only be visible
	// if AllDecls is set.
	error
}

// This error declaration shadows the predeclared error type.
type error interface {
	Error() string
}
