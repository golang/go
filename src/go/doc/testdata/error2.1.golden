// 
PACKAGE error2

IMPORTPATH
	testdata/error2

FILENAMES
	testdata/error2.go

TYPES
	// 
	type I0 interface {
		// When embedded, the locally-declared error interface
		// is only visible if all declarations are shown.
		error
	}

	// 
	type S0 struct {
		// In struct types, an embedded error must only be visible
		// if AllDecls is set.
		error
	}

	// 
	type T0 struct {
		ExportedField interface {
			// error should not be visible
			error
		}
	}

	// This error declaration shadows the predeclared error type. 
	type error interface {
		Error() string
	}

